internal/service/resourcepolicy/data_source.go

```diff
@@ -0,0 +1,53 @@
+package resourcepolicy
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+)
+
+var _ datasource.DataSource = &resourcePolicyDS{}
+var _ datasource.DataSourceWithConfigure = &resourcePolicyDS{}
+
+const (
+	errorReadDS = "error reading data source " + fullResourceName
+)
+
+func DataSource() datasource.DataSource {
+	return &resourcePolicyDS{
+		DSCommon: config.DSCommon{
+			DataSourceName: resourceName,
+		},
+	}
+}
+
+type resourcePolicyDS struct {
+	config.DSCommon
+}
+
+func (d *resourcePolicyDS) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = DataSourceSchema(ctx)
+}
+
+func (d *resourcePolicyDS) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var cfg TFResourcePolicyModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &cfg)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := d.Client.AtlasV2
+	apiResp, _, err := connV2.AtlasResourcePoliciesApi.GetAtlasResourcePolicy(ctx, cfg.OrgID.ValueString(), cfg.ID.ValueString()).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError(errorReadDS, err.Error())
+		return
+	}
+
+	out, diags := NewTFResourcePolicyModel(ctx, apiResp)
```

Comment:
knit: what about name it tfModel instead of out?

Reply:
Thanks. Updated in c2ee9efa28658d86eaff5155f0ef5b0f05f1b1af